3rdpartylibrary/google-maps-android-cmgn/src/main/java/com/google/maps/android/kml/KmlFeatureParser.java

```diff
@@ -132,15 +135,23 @@ private static float getRotation(XmlPullParser parser)
     /**
      * Retrieves a url from the "href" tag nested within an "Icon" tag, read by
      * the XmlPullParser.
+     * If the image is a local reference, get the absolute path from the documentBase
+     * and then cast it into a File URl.
      *
+     * @param documentBase Path leading to kml file directory to generate full path file url for map service.
+     *                     For example if the file was in C:/Documents/MyKMZ.kmz/test.kml this value would be C:/Documents/MyKMZ.kmz
      * @return An image url
      */
-    private static String getImageUrl(XmlPullParser parser)
+    private static String getImageUrl(XmlPullParser parser, String documentBase)
             throws IOException, XmlPullParserException {
         int eventType = parser.getEventType();
         while (!(eventType == XmlPullParser.END_TAG && parser.getName().equals("Icon"))) {
             if (eventType == XmlPullParser.START_TAG && parser.getName().equals("href")) {
-                return parser.nextText();
+                String imageUrl =  parser.nextText();
```

Comment:
I forgot to mention this before but please add the word final before all variables